app/forms/questionnaire_form.py

```diff
@@ -203,27 +204,32 @@ def _get_target_total_and_currency(
         self,
         calculation: Calculation,
         question: QuestionSchemaType,
-    ) -> Optional[tuple[Union[Calculation, AnswerValueTypes], Optional[str]]]:
-        calculation_value: Union[Calculation, AnswerValueTypes]
-        currency: Optional[str]
+    ) -> dict:
+        result = {}
 
         if "value" in calculation:
             if isinstance(calculation["value"], dict):
-                calculation_value = self.value_source_resolver.resolve(calculation["value"])  # type: ignore
+                result["target_total"] = self.value_source_resolver.resolve(calculation["value"])  # type: ignore
             else:
-                calculation_value = calculation["value"]
-            currency = question.get("currency")
-            return calculation_value, currency
+                result["target_total"] = calculation["value"]
+            result["currency"] = question.get("currency")
+            result["decimal_places"] = self.schema.get_decimal_limit(
+                calculation["answers_to_calculate"]
+            )
```

Comment:
If you do this at the very start, you don't need to duplicate it below for non value one.

Reply:
Updated 👍